lua/lualine.lua

```diff
@@ -150,32 +152,60 @@ local function statusline(sections, is_focused)
   if M.theme ~= theme_set then
     set_lualine_theme()
   end
-  local status = {}
-  if sections.lualine_a then
-    local hl = highlight.format_highlight(is_focused, 'lualine_a')
-    table.insert(status, utils_component.draw_section(sections.lualine_a,  hl))
-  end
-  if sections.lualine_b then
-    local hl = highlight.format_highlight(is_focused, 'lualine_b')
-    table.insert(status, utils_component.draw_section(sections.lualine_b,  hl))
-  end
-  if sections.lualine_c then
-    local hl = highlight.format_highlight(is_focused, 'lualine_c')
-    table.insert(status, utils_component.draw_section(sections.lualine_c,  hl))
-  end
-  table.insert(status, "%=")
-  if sections.lualine_x then
-    local hl = highlight.format_highlight(is_focused, 'lualine_c')
-    table.insert(status, utils_component.draw_section(sections.lualine_x,  hl))
-  end
-  if sections.lualine_y then
-    local hl = highlight.format_highlight(is_focused, 'lualine_b')
-    table.insert(status, utils_component.draw_section(sections.lualine_y,  hl))
+  -- status_builder stores statusline without section_separators
+  local status_builder = {}
+  -- The sequence sections should maintain
+  local section_sequence = {'a', 'b', 'c', 'x', 'y', 'z'}
+
+  for _, sec in ipairs(section_sequence) do
+    if sections['lualine_'..sec] then
+      -- insert highlight+components of this section to status_builder
+      local highlight = highlight.format_highlight(is_focused,
+                   'lualine_'..sec)
+			local section = utils_component.draw_section(sections['lualine_'..sec], highlight)
+			if #section > 0 then
+				table.insert(status_builder, {sec, section})
+			end
+    end
   end
-  if sections.lualine_z then
-    local hl = highlight.format_highlight(is_focused, 'lualine_a')
-    table.insert(status, utils_component.draw_section(sections.lualine_z,  hl))
+
+  -- Actual statusline
+  local status = {}
+  local half_passed = false
+  for i=1,#status_builder do
+    -- midsection divider
+    if not half_passed and status_builder[i][1] > 'c' then
+      table.insert(status, highlight.format_highlight(is_focused, 'lualine_c') .. '%=')
+      half_passed = true
+    end
+    -- provide section_separators when statusline is in focus
+    if is_focused then
+      -- component separator needs to have fg = current_section.bg
+      -- and bg = adjacent_section.bg
+      local prev_section = status_builder[i-1] and status_builder[i-1] or {[2] = nil}
```

Comment:
I think it would be cleaner to use a map rather than array here. `[2]=nil]` is hard to read.

Reply:
does {nil, nil} seem better?